### PR TITLE
[react-slider] Creates generic and updates types

### PR DESCRIPTION
**Pull request by waynevanson**
There were two render functions missing alongside a missing generic, so there is no error handling for `typeof newState === 'number'` required when using the `onChange` handler.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
+ <https://zillow.github.io/react-slider/>
+ <https://github.com/zillow/react-slider/blob/master/src/components/ReactSlider/ReactSlider.jsx#L74-L260>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/react-slider/index.d.ts [hunk 1]
```diff
@@ -1,36 +1,140 @@
-// Type definitions for react-slider 0.8
+// Type definitions for react-slider 1.0
 // Project: https://github.com/mpowaga/react-slider
 // Definitions by: Jason Unger <https://github.com/jsonunger>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+//                 Wayne Van Son <https://github.com/waynevanson>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped/types/react-slider
+// TypeScript Version: 3.6
```
Comment (sandersn) [inline types/react-slider/index.d.ts]: What features of typescript 3.6 does this require? Moving from 2.8 to 3.6 means that this update will only be available by default to people on 3.6 or higher. Everybody else will be stuck with the old version.

File: types/react-slider/react-slider-tests.tsx [hunk 2]
```diff
@@ -1,15 +1,18 @@
 import * as React from 'react';
 import ReactSlider from 'react-slider';
```
Comment (sandersn) [inline types/react-slider/react-slider-tests.tsx]: Use the commonjs syntax `import ReactSlider = require('react-slider')`, and change `"esModuleInterop": false` in tsconfig.

Consumers of the package who want to use `import ReactSlider from 'react-slider'` will need to add `"esModuleInterop": true`, but Definitely Typed tests need to be backward compatible to show that the types will work with old code bases.

**Review by sandersn — CHANGES_REQUESTED** [review]

**Comment by waynevanson** [comment]
Sorry about this! I'll fix it up now. Thank you for your feedback.

**Comment by waynevanson** [comment]
I'm having an error with the react imports.

I used this import syntax
```js
import React = require('react');
```
And this is what I got when running `npm run lint react-slider`:

``` terminal
Error: Errors in typescript@next for external dependencies:
../react/index.d.ts(33,22): error TS2307: Cannot find module 'csstype'.
```

If @bjorgvin is still working then we may as well just use his one.

**Pull request closed without merging**